src/ml_tooling/baseclass.py

```diff
@@ -198,6 +195,25 @@ def make_prediction(self, input_data, proba=False, use_index=False) -> pd.DataFr
             message = f"You haven't fitted the model. Call 'train_model' or 'score_model' first"
             raise MLToolingError(message) from None
 
+    @property
+    def default_metric(self):
+        """
+        Finds estimator_type for estimator in a BaseClassModel and returns default
+        metric for this class stated in .config. If passed estimator is a Pipeline,
+        assume last step is the estimator.
+
+        Returns
+        -------
+        str
+            Name of the metric
+
+        """
+        estimator_type = self.model._estimator_type
```

Comment:
Would do this as a oneliner - 

```python
return  self.config.CLASSIFIER_METRIC if self.model._estimator type == 'classifier'  else self.config.REGRESSION_METRIC
```

a bit more pythonic

Reply:
Would be a nice feature to have a property.setter as well, so the user can change either .default_metric or .config.*_METRIC. The setter would then update the correct *_METRIC value based on model estimator type

Reply:
I have added now with tests.